### PR TITLE
charts draggable

### PR DESCRIPTION
**Pull request by akash-groww**
## What does this PR do?

Implements draggable feature in LineGraph

## What packages have been affected by this PR?
react-charts
base-css

## Types of changes

What types of changes does your code introduce to this project?

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist before merging

_Put an `x` in the boxes that apply_

- [ ] These changes have been thoroughly tested.

- [ ] Changes need to be immediately published on npm.

### CODE REVIEW

File: packages/react-charts/src/LineGraph/lineGraphTypes.tsx [hunk 1]
```diff
@@ -71,3 +84,8 @@ export type ToolTipData = {
   seriesData: ToolTipSeriesData[];
   otherData: HoveredPointData;
 }
+
+export type DragData = {
+  startPoint: number;
+  endPoint?: number;
```
Comment (samyak0472) [inline packages/react-charts/src/LineGraph/lineGraphTypes.tsx:90]: startIndex endIndex

File: packages/react-charts/src/LineGraph/lineGraphTypes.tsx [hunk 2]
```diff
@@ -39,14 +45,19 @@ export type LineGraphProps = {
 
   onMouseEnter?: (td: ToolTipData) => void; //on mouse enter callback
   onMouseLeave?: ()=> void;        // on Mouse leave callback
+  onMouseUp?: ()=> void;        // on Mouse Up callback
+  onMouseDown?: ()=> void;        // on Mouse Down callback
 
   toolTipLeftUpdated?: number;   // calculated/updated toolTipLeft
   toolTipTopUpdated?: number;    // calculated/updated toolTipTop
   getTooltipUI?: (toolTipData: ToolTipData) => JSX.Element; //get tool tip ui
+  getDefs?: () => JSX.Element; //get defs for ui
```
Comment (samyak0472) [inline packages/react-charts/src/LineGraph/lineGraphTypes.tsx:54]: Return Type should be SVGProps.<something>

File: packages/react-charts/src/LineGraph/lineGraphTypes.tsx [hunk 3]
```diff
@@ -59,6 +70,8 @@ export type ToolTipSeriesData = {
   tooltipTop: number;
   isPerfectIntersection: boolean;
   intersectionPointOnLine: HoveredPointData;
+  prevPoint?: Point;
```
Comment (samyak0472) [inline packages/react-charts/src/LineGraph/lineGraphTypes.tsx:73]: dragStartPoint

File: packages/react-charts/src/LineGraph/lineGraphTypes.tsx [hunk 4]
```diff
@@ -59,6 +70,8 @@ export type ToolTipSeriesData = {
...
+  prevTooltipLeft?: number;
```
Comment (samyak0472) [inline packages/react-charts/src/LineGraph/lineGraphTypes.tsx:74]: same

File: packages/react-charts/src/LineGraph/index.tsx [hunk 5]
```diff
@@ -171,7 +229,21 @@ const LineGraph = (props: LineGraphProps) => {
...
+  const getDragLinePoints = (series: Point[]) => {
+    let { startPoint, endPoint } = dragPoints ?? {};
+
+    if (startPoint === undefined || endPoint === undefined) return [];
+    if (startPoint > endPoint) {
+      [ startPoint, endPoint ] = [ endPoint, startPoint + 1 ];
 
+    } else {
+      endPoint += 1;
+    }
```
Comment (samyak0472) [inline packages/react-charts/src/LineGraph/index.tsx:244]: if (startPoint > endPoint) {
      [ startPoint, endPoint ] = [ endPoint, startPoint ];

    } 
      endPoint += 1;

File: packages/react-charts/src/LineGraph/index.tsx [hunk 6]
```diff
@@ -107,9 +109,43 @@ const LineGraph = (props: LineGraphProps) => {
...
+    const { fill = lp?.color, lineColor, negativeLineColor } = lp?.draggableConfig ?? {};
+
+    if (startPoint === undefined || endPoint === undefined) return lp?.color;
+    const lpStart = lp?.series?.[startPoint]?.[1];
+    const lpEnd = lp?.series?.[endPoint]?.[1];
+    const line = lineColor ?? fill;
+    const negativeLine = negativeLineColor ?? lineColor ?? fill;
+
+    if (startPoint < endPoint) return lpStart < lpEnd ? line : negativeLine;
+    return lpStart > lpEnd ? line : negativeLine;
```
Comment (samyak0472) [inline packages/react-charts/src/LineGraph/index.tsx:123]: simplify logic

**Review by samyak0472 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by harshitpruthi94 — APPROVED** [review]

**Review by harshitpruthi94 — APPROVED** [review]

**Pull request merged**